flink/src/main/java/org/apache/iceberg/flink/sink/IcebergStreamWriter.java

```diff
@@ -62,7 +62,7 @@ public void open() {
   @Override
   public void prepareSnapshotPreBarrier(long checkpointId) throws Exception {
     // close all open files and emit files to downstream committer operator
-    for (DataFile dataFile : writer.complete()) {
+    for (DataFile dataFile : writer.complete().dataFiles()) {
```

Comment:
Until support is added to handle delete files, I think it would make sense to add a precondition to check that there are no delete files here.

Reply:
Actually, it looks like this change happened in quite a few places. What do you think about returning `DataFile[]` from a new `dataFiles()` method and doing the precondition check for 0 delete files in that method?

Then we could update all of the places that call `complete().dataFiles()` to just `dataFiles()` and ensure that the check is done whenever `dataFiles()` is called to get the result rather than `complete`.

Reply:
I like the idea about introducing a new `dataFiles` in `TaskWriter` rather than precondition check everywhere.